### PR TITLE
More useful installer

### PR DESCRIPTION
**Pull request by rmg**
Adds functionality that is common to the existing strong-pm installer and the to-be-implemented strong-central and strong-executor installers.

Connect to strongloop-internal/scrum-nodeops#681
Connect to strongloop-internal/scrum-nodeops#682

### CODE REVIEW

**Comment by rmg** [comment]
@slnode test please

**Comment by rmg** [comment]
@slnode test please

**Comment by rmg** [comment]
@sam-github to reduce the amount for you to review, I merged all the trivial test and dependency updates

**Comment by rmg** [comment]
@slnode test please

File: index.js [hunk 1]
```diff
@@ -183,32 +241,231 @@ function ensureJobFileDir(opts, cb) {
...
+function ensureUser(opts, callback) {
+  userExists(opts.user, function(err, exists) {
+    if (err || exists)
+      return callback(err);
+    if (opts.dryRun) {
+      install.log('skipping user creation in dry-run');
+      return callback();
+    }
+    if (install.platform !== 'linux') {
+      install.log('skipping user creation on non-Linux platform');
```
Comment (sam-github) [inline index.js]: does any element of the installation work on non-Linux?

**Comment by sam-github** [comment]
LGTM

**Pull request merged**